### PR TITLE
feat(dashboard): skeletons for repositories and synced sandboxes

### PR DESCRIPTION
**Pull request by olarclara**
## What kind of change does this PR introduce?

Updates the loading UI for repositories and synced sandboxes. There are three types of skeletons right now:

- Default: serves sandboxes, cloud sandboxes, templates, and branches.
- Solid: serves repositories and synced sandboxes.

## What is the new behavior?

|Repositories|Synced sandboxes|
|:-:|:-:|
|localhost_3000_dashboard_repositories_workspace=642af90c-4717-4730-bad3-e4c1e37ca5e2|localhost_3000_dashboard_synced-sandboxes_workspace=642af90c-4717-4730-bad3-e4c1e37ca5e2|

### CODE REVIEW

**Comment by lbogdan** [comment]
Build for latest commit https://github.com/codesandbox/codesandbox-client/pull/7017/commits/3e36c453a2c46d25f4bc05a5ccc83254c3dcc872 is at https://pr7017.build.csb.dev/s/new.

File: packages/app/src/app/pages/Dashboard/Components/Skeleton/DefaultSkeleton.tsx [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  const { dashboard } = useAppState();
+  const location = useLocation();
+
+  let viewMode;
+  if (location.pathname.includes('archive')) viewMode = 'list';
+  else viewMode = dashboard.viewMode;
+
+  if (viewMode === 'list') {
+    return <DefaultSkeletonListItem />;
+  }
```
Comment (tristandubbeld) [inline packages/app/src/app/pages/Dashboard/Components/Skeleton/DefaultSkeleton.tsx:60]: We could keep this logic out of the `Skeleton` component and add it as a prop when necessary, or would that be a lot of extra code or inconvenience in this case?

**Review by tristandubbeld — APPROVED** [review]
Skeletons look good! One comment about moving logic out of the Skeleton component. Curious what you think!

File: packages/app/src/app/pages/Dashboard/Components/Skeleton/DefaultSkeleton.tsx (hunk 1 above)
Comment (olarclara) [reply packages/app/src/app/pages/Dashboard/Components/Skeleton/DefaultSkeleton.tsx]: @tristandubbeld great point! replaced the logic with a prop here: [be43212](https://github.com/codesandbox/codesandbox-client/pull/7017/commits/be43212bc9ad7b7058db19ba0ced77be1d1c79ba).

**Pull request merged**